### ISSUE TITLE
file is deleted when opening downloaded file while offline

### ISSUE THREAD

**Issue by tessus** [nextcloud/android#3642]
### Actual behaviour
File is deleted

### Expected behaviour
File should be shown
 
### Steps to reproduce
- Open Nextcloud app on Android
- Download a PDF file (opening the PDF works fine)
- Turn off mobile data and Wifi
- Open the PDF
- Message appears: `File could not be synced. Showing latest available version.`
- But instead of showing the file, it is deleted.

### Environment data
Android version: 9

Device model: Google Pixel

Stock or customized system: Stock

Nextcloud app version: 3.5.0

Nextcloud server version: 14.0.6

#### Web server error log
```
no log entries (there can't be any entries: the client has no data connection)
```

#### Nextcloud log (data/nextcloud.log)
```
no log entries (there can't be any entries: the client has no data connection)
```

**Comment by tanvir1414** [comment]
Same problem. It is true for all file types. Also "set as offline" is missing.

**Comment by tessus** [comment]
@tanvir1414 the PDF file was just an example... ;-)

**Comment by topas-rec** [comment]
Duplicate of #3638 ?

**Comment by tessus** [comment]
@topas-rec well, I figured out in which edge case the data loss occurs, so my issue is actually the correct one. But I don't care which one is closed, as long as it is fixed.

I lost my boarding pass 2 days ago at the airport, so yes, this is rather serious. The moment you need an offline file most is the time it vanishes into thin air....

**Comment by topas-rec** [comment]
> well, I figured out in which edge case the data loss occurs, so my issue is actually the correct one.

Good point. I thought older is better in case of github isues. ;-)

Main thing is it gets fixed. Wanted to go by train. Credit card things got lost by loosing my keepass store. I was able to fire up my pc cause there was enough time but it might also have been too late. ( Well there is still the hardware credit card yes) Your situation sounds even worse!

**Comment by tessus** [comment]
I suggest @AndyScherzinger decides which issue will prevail. In any case, now he has the steps to reproduce the issue.

**Comment by AndyScherzinger** [comment]
@tessus I'll leave that up to @tobiasKaminsky since he rather has the time to fix this than I have...

**Comment by jgoerzen** [comment]
Not entirely sure if this is the same issue, but I had the same thing happen multiple times while in places with marginal connectivity -- it would say "Fetching latest copy from server" for about 20 seconds, then fail, delete the file, and leave me high and dry.  Quite problematic as this was also travel-related.

**Comment by ndehav01** [comment]
Nextcloud App: 3.5.0
Android: 6.0 (LG G3)

I've been experiencing the same problem described in the OP for a week or more.

Files downloaded while data or wi-fi connections are accessible through the app, but as soon as signal is lost and the phone gives the usual message: "File could not be synced. Showing latest available version." And the file is subsequently deleted.

There was previously also an option for files that allowed you to "Set as available offline", which is now gone, replaced by just  a "Download" or "Sync". No file deletions have synced back to the server causing data loss, but that could be because I had opted for read-only. I don't see that option now either.

I am working around the issue now by using a link to the actual downloaded media folder to access files directly, avoiding the Nextcloud app.

**Comment by Buntbart** [comment]
Same problem here. Samsung Galaxy S5 (klte), LineageOS 15.1 (Android 8.1). 
Do you plan to fix this asap or should I look for another solution. My need for a cloud-solution that deletes my files is pretty small.

**Comment by AndyScherzinger** [comment]
Like stated in https://github.com/nextcloud/android/issues/3642#issuecomment-464867275 I am forced to leave the fixing part to @tobiasKaminsky since my day job is taking away most of my time at this very moment so I sstill hope that @tobiasKaminsky will find the time to fix this.

**Comment by tessus** [comment]
I hope so too. This is a rather nasty bug with data loss involved. Well, you can get it back when you're back online, but at the moment of need the data is gone...

Anyway, I would have imagined that an emergency release be created right away. Or a rollback of the previous release initiated. Neither happened.
I really think you should start working on contingency plans. For the lack of a better term, the current release management sucks.

(Now I'm going to be mean - although I have mentione many, many times that priorities are way off in the nextcloud community: maybe a button needs tweaking first...)

**Comment by topas-rec** [comment]
There is a simple thing you can do yourself to get a release without this bug by reverting to the previous version. (I assume the old version still works with nextclouds current server release)

Another way to work around this issue is simply using a file manager for accessing the files.

Bad thing that persists is: new users that experiment with nextcloud ecosystem might be frustrated by this issue and might stay away from nc.

**Comment by tessus** [comment]
> There is a simple thing you can do yourself to get a release without this bug by reverting to the previous version. (I assume the old version still works with nextclouds current server release)

Yes, please, tell me how I can instruct the Google Play Store to install the previous version.

> Another way to work around this issue is simply using a file manager for accessing the files.

I'm aware of several workarounds but that's not the issue here.

**Comment by AndyScherzinger** [comment]
> Yes, please, tell me how I can instruct the Google Play Store to install the previous version.

you can't as far as I know. The only way I know of is to uninstall/re-install while you'd then used the signed gplay apk version from the tag _and_ need to tell gplay to not update the app through the app store since it will then offer you 3.5.0 since it detectes the device not running the latest version known to gplay.
https://github.com/nextcloud/android/releases/tag/stable-3.4.2

**Comment by topas-rec** [comment]
Ah I see. I use https://f-droid.org/ next to Google play store, too.

There one can select the version that should be installed (including older versions).

I did not test this into depth because simple downgrading dues not work (error), but uninstalling the app and reinstalling an older version by selecting it works. (Tested it with another app just now)

I'm just too lazy to do it. I use an alternative for explorer.

**Comment by tessus** [comment]
> Ah I see. I use https://f-droid.org/ next to Google play store, too.

Yea, I did that too, until I got a message from the nextcloud client that something is not working in the f-droid version (I believe it was notifications). Then I switched to the Google Play Store version.

Also, afaik installing the f-droid version or the apk would delete my preferences and I'd have to setup from scratch. 

I'm currently just not using it to read offlne files. That's it. Still a bugger that there's no emergency update. Anyway, I was right. It's more important to fix code indentation f4f53cf254d4ea3d5a99c95be5797dce9f1fc0f5 than this bug.

Well, luckily my life doesn't depend on Nextcloud, otherwise I'd be dead already. But what can one expect from a free product where devs constantly tell you that priorities are set according to who pays most.
Anyway, I opened the issue. They can leave it or take it. I don't care anymore.
